docs/reference/settings.md

```diff
@@ -619,6 +619,39 @@ bypasses SSL verification and could expose you to MITM attacks.
 
 ---
 
+### [`bounds`](#bounds) {: #bounds }
+
+The default version specifier when adding a dependency.
+
+If no constraint or URL is provided for a dependency, a bound is added based on the
+latest compatible version of the package, e.g., `>=1.2.3`, unless `--frozen` is provided, in
+which case no resolution is performed.
+
+**Default value**: `"lower"`
+
+**Possible values**:
+
+- `"lower"`: Only a lower bound, e.g., `>=1.2.3`
+- `"major"`: Allow the same major version, similar to the semver caret, e.g., `>=1.2.3,<2.0.0`
+- `"minor"`: Allow the same minor version, similar to the semver tilde does, `>=1.2.3,<1.3.0`
+- `"exact"`: Pin the exact version, e.g., `==1.2.3`
+
+**Example usage**:
+
+=== "pyproject.toml"
+
+    ```toml
+    [tool.uv]
+    bounds = "major"
```

Comment:
I worry this feels like we'll use these bounds in other operations. I wonder if it should be `default-bounds`? or `add.bounds`? Do you think we'll read this setting during `uv upgrade` eventually? Would we want a separate option for that?

Reply:
(`edit.bounds` is okay, but I worry it won't be obvious to users and it _only_ applies to `add` edits, right?)

Reply:
`add.bounds` sounds good. I have to do the systematic research for that still, but I expect upgrade to have more complex options, such as https://docs.github.com/en/code-security/dependabot/working-with-dependabot/dependabot-options-reference#versioning-strategy--.